modules/metrics/prometheus/src/main/scala/higherkindness/mu/rpc/prometheus/PrometheusMetrics.scala

```diff
@@ -16,12 +16,172 @@
 
 package higherkindness.mu.rpc.prometheus
 
-import io.prometheus.client.CollectorRegistry
+import cats.effect.Sync
+import cats.syntax.functor._
+import higherkindness.mu.rpc.internal.interceptors.GrpcMethodInfo
+import io.prometheus.client._
 import higherkindness.mu.rpc.internal.metrics.MetricsOps
+import higherkindness.mu.rpc.internal.metrics.MetricsOps._
+import io.grpc.Status
+
+/*
+ * [[MetricsOps]] algebra able to record Prometheus metrics.
+ *
+ * The list of registered metrics contains:
+ *
+ * {prefix}_active_calls{labels=classifier} - Counter
+ * {prefix}_messages_sent{labels=classifier,service,method} - Counter
+ * {prefix}_messages_received{labels=classifier,service,method} - Counter
+ * {prefix}_calls_header{labels=classifier} - Histogram
+ * {prefix}_calls_total{labels=classifier} - Histogram
+ * {prefix}_calls_by_method{labels=classifier,method} - Histogram
+ * {prefix}_calls_by_method{labels=classifier,status} - Histogram
```

Comment:
I think we should combine these two metrics into a single one, just adding the three labels (`classifier`, `method`, and `status`)